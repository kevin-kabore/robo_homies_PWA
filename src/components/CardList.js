import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
	console.log('CardList')
  return (
    <div>
      {robots.map((robot, i) => {
        return (
          <Card key={i} id={robot.id} name={robot.name} email={robot.email} />
        );
      })}
    </div>
  );
};

export default CardList;
