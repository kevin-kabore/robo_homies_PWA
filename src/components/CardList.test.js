import { shallow } from 'enzyme';
import React from 'react';

import CardList from './CardList';

describe('CardList Component', () => {
	const mockRobots = [
	{
		id: 1,
		name: 'John Snow',
		email: 'johm.snow@gmail.com'
	}]
	it('Will render the Component', () => {
		// snapshot testing used for all presentational/stateless components the whole component
		expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot()		
	})
})
