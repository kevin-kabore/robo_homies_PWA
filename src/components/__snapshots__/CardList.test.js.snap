// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component Will render the Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "johm.snow@gmail.com",
              "id": 1,
              "name": "John Snow",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="johm.snow@gmail.com"
          id={1}
          name="John Snow"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "johm.snow@gmail.com",
          "id": 1,
          "name": "John Snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="johm.snow@gmail.com"
            id={1}
            name="John Snow"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "johm.snow@gmail.com",
            "id": 1,
            "name": "John Snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
